python/scannerpy/database.py

```diff
@@ -1495,30 +1459,14 @@ def run(self,
                     sink_args = j.outputs.add()
                     sink_args.op_index = op_idx
 
-                    if not op in op_to_op_args:
-                        raise ScannerException(
-                            'No arguments bound to sink {:s}.'.format(
-                                op._name))
-                    else:
-                        args = op_to_op_args[op]
+                    args = op_to_op_args[op]
+
+                    sink_args.args = args
+
+                    streams = op._streams
+                    storage = streams[0].storage()
+                    storage.delete(self, [s for s in streams if s.exists()])
```

Comment:
Force deleting everything?

Reply:
Added a `cache_mode` mechanism. What do you think?

Reply:
Looks good to me!